simphony/cuds/mesh.py

```diff
@@ -218,13 +218,26 @@ class Mesh(ABCMesh):
 
     def __init__(self, name):
         self.name = name
-        self.data = 0
 
         self._points = {}
         self._edges = {}
         self._faces = {}
         self._cells = {}
 
+        self._data = dc.DataContainer()
+
+    @property
+    def data(self):
+        return self._data
+
+    @data.setter
+    def data(self, value):
+        if not isinstance(value, dc.DataContainer):
```

Comment:
I do no think that there is a need for checking the type (for example data containers coming from adapters will not necessarily be subclasses of DataContainer). As long as the `value` is a CUBA based mapping it should be good for us. I suggest instead to do the check indirectly as:
```
self._data = DataContainer(value)
```
Which will also make sure that we make a copy.

In general in pyhon specific type checking is avoided unless is absolutely essential to the operation
which will make sure that